### PR TITLE
Spark 3.1.2 support

### PR DESCRIPTION
**Pull request by arielshaqed**
This is currently suitable *only* for internal use; do *NOT* merge to
master!

### CODE REVIEW

**Comment by arielshaqed** [comment]
Tested on EMR with known issues (WIP).  Currently pushed (only) to internal S3 buckets for use.  However this _does_ load, and works to a point.

# DO NOT MERGE

(until further notice)

**Review by talSofer — COMMENTED** [review]
Thanks @arielshaqed!

**Review by talSofer — COMMENTED** [review]
WOW, amazing work!

File: clients/spark/build.sbt [hunk 1]
```diff
@@ -126,7 +133,7 @@ lazy val s3UploadSettings = Seq(
     (assemblyOutputPath in assembly).value ->
       s"${name.value}/${version.value}/${(assemblyJarName in assembly).value}"
   ),
-  s3Upload / s3Host := "treeverse-clients-us-east.s3.amazonaws.com",
+  s3Upload / s3Host := "bobotreeverse-clients-us-east.s3.amazonaws.com",
```
Comment (Jonathan-Rosenberg) [inline clients/spark/build.sbt:136]: `bobotreeverse`?

**Review by Jonathan-Rosenberg — APPROVED** [review]
LGTM except for the use of Scala 😜

**Comment by arielshaqed** [comment]
# Merging but not to trunk!

Thanks...

**Pull request merged**